packages/expo-audio/src/Audio.types.ts

```diff
@@ -275,6 +275,36 @@ export enum AudioQuality {
  */
 export type BitRateStrategy = 'constant' | 'longTermAverage' | 'variableConstrained' | 'variable';
 
+/**
+ * Options for controlling how audio recording is started.
+ */
+export type RecordingStartOptions = {
+  /**
+   * The duration in seconds after which recording should automatically stop.
+   * If not provided, recording continues until manually stopped.
+   *
+   * @platform ios
+   * @platform android
+   * @platform web
+   */
+  forDuration?: number;
+  /**
+   * The time in seconds to wait before starting the recording.
+   * If not provided, recording starts immediately.
+   *
+   * **Platform behavior:**
+   * - iOS: Uses native AVAudioRecorder.record(atTime:) for precise timing.
+   *   ⚠️ **Important**: The recording process starts immediately (you'll see status updates),
+   *   but actual audio capture begins after the specified delay. This is not a countdown -
+   *   the recorder is active but silent during the delay period.
+   * - Android: Ignored, recording starts immediately
+   * - Web: Ignored, recording starts immediately
```

Comment:
```suggestion
   * **Platform behavior:**
   * - Android: Ignored, recording starts immediately
   * - iOS: Uses native AVAudioRecorder.record(atTime:) for precise timing.
   * - Web: Ignored, recording starts immediately
   *
   * > **warning** On iOS, the recording process starts immediately (you'll see status updates),
   * but actual audio capture begins after the specified delay. This is not a countdown, since
   * the recorder is active but silent during the delay period.
```

Let's use the platform order as per the writing style guide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#platform-order-when-referencing-android-ios-and-web-or-multiple-platforms

I'd avoid the emoji pattern you are using to call out the warning. Let's use the standard callout pattern after the platform behavior section and use `> **warning**` instead of important (we use important for experimental feature indication mostly). If this warning is only related to iOS, in the callout, let's mention the platform name.

Reply:
Also, since this marked as platform iOS only, I think Android and Web info might not be required.